vehicles/src/modules/credit-account/profiles/credit-account.profile.ts

```diff
@@ -0,0 +1,149 @@
+import { AutomapperProfile, InjectMapper } from '@automapper/nestjs';
+import {
+  Mapper,
+  createMap,
+  forMember,
+  forSelf,
+  fromValue,
+  mapFrom,
+  mapWithArguments,
+} from '@automapper/core';
+import { Injectable } from '@nestjs/common';
+import { CreditAccountUser } from '../entities/credit-account-user.entity';
+import { CreateCreditAccountUserDto } from '../dto/request/create-credit-account-user.dto';
+import { ReadCreditAccountUserDto } from '../dto/response/read-credit-account-user.dto';
+import { Company } from '../../company-user-management/company/entities/company.entity';
+import { CreditAccountUserType } from '../../../common/enum/credit-accounts.enum';
+import { CreditAccount } from '../entities/credit-account.entity';
+import { ReadCreditAccountDto } from '../dto/response/read-credit-account.dto';
+import { CreditAccountLimitType } from '../../../common/enum/credit-account-limit.enum';
+
+@Injectable()
+export class CreditAccountProfile extends AutomapperProfile {
+  constructor(@InjectMapper() mapper: Mapper) {
+    super(mapper);
+  }
+
+  override get profile() {
+    return (mapper: Mapper) => {
+      createMap(
+        mapper,
+        CreditAccount,
+        ReadCreditAccountDto,
+        forMember(
+          (d) => d.companyId,
+          mapWithArguments((_s, { companyId }) => companyId as number),
```

Comment:
Change mapping to mapFrom as companyId is already available in the source. External mapping is not required